src/quest/serializer.py

```diff
@@ -1,13 +1,41 @@
-from typing import Any, Dict, Tuple
-import importlib
+from typing import Any, Dict, Protocol, TypeVar, Callable
+from utils import get_obj_name, import_object
 
-from .utils import get_obj_name, import_object
+T = TypeVar('T')
 
 
-class MasterSerializer:
-    def __init__(self, type_serializers=None):
-        # Initialize with a dictionary of custom serializers - default to empty dict
+class StepSerializer(Protocol):
```

Comment:
Look at the signatures I put in the scratch file. Note how the generic var `T` is used.

Reply:
in fact, looking at the usage below, I think we want something like this:
```py
class TypeFactory(Protocol[T]):
    async def __call__(self, *args, **kwargs) -> T: ...

class TypeSerializer(Protocol[T]):
    async def serialize(self, obj: T) -> tuple[TypeFactory[T], tuple, dict]: ...
```

then below you can delete `_get_serializer_for_factory` and instead just load the factory from the name/module stored in the dictionary.

We may need a check to ensure that the deserializer provided is not a lambda, as you can't dynamically import lambdas. If they try to use a lambda, we should throw a ValueError.